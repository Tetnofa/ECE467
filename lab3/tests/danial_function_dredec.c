int main();
int main() {

}
int main() { 

}
int main() {

}
int main() {

}
void helper1(int x);
int helper1(int y);
void helper1(int x) {

}
int helper(bool x) {

}
int f(int x, int y, int z);

int f(bool x, int y, int z) {

}
int poo () {

}
int poo () {

}
void poo () {

}
bool poo () {

}
// int f(int z) {

// }