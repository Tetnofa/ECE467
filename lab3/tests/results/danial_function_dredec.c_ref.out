Program [useIO=0](1,0)
	Function[isProto=1]	Int(1,0)
		Identifier[name:main](1,4)
	Function[isProto=0]	Int(2,0)
		Identifier[name:main](2,4)
		Scope(2,11)
	Function[isProto=0]	Int(5,0)
		Identifier[name:main](5,4)
		Scope(5,11)
	Function[isProto=0]	Int(8,0)
		Identifier[name:main](8,4)
		Scope(8,11)
	Function[isProto=0]	Int(11,0)
		Identifier[name:main](11,4)
		Scope(11,11)
	Function[isProto=1]	Void(14,0)
		Identifier[name:helper1](14,5)
		Parameter		Int(14,13)
			Identifier[name:x](14,17)
	Function[isProto=1]	Int(15,0)
		Identifier[name:helper1](15,4)
		Parameter		Int(15,12)
			Identifier[name:y](15,16)
	Function[isProto=0]	Void(16,0)
		Identifier[name:helper1](16,5)
		Parameter		Int(16,13)
			Identifier[name:x](16,17)
		Scope(16,20)
	Function[isProto=0]	Int(19,0)
		Identifier[name:helper](19,4)
		Parameter		Bool(19,11)
			Identifier[name:x](19,16)
		Scope(19,19)
	Function[isProto=1]	Int(22,0)
		Identifier[name:f](22,4)
		Parameter		Int(22,6)
			Identifier[name:x](22,10)
		Parameter		Int(22,13)
			Identifier[name:y](22,17)
		Parameter		Int(22,20)
			Identifier[name:z](22,24)
	Function[isProto=0]	Int(24,0)
		Identifier[name:f](24,4)
		Parameter		Bool(24,6)
			Identifier[name:x](24,11)
		Parameter		Int(24,14)
			Identifier[name:y](24,18)
		Parameter		Int(24,21)
			Identifier[name:z](24,25)
		Scope(24,28)
	Function[isProto=0]	Int(27,0)
		Identifier[name:poo](27,4)
		Scope(27,11)
	Function[isProto=0]	Int(30,0)
		Identifier[name:poo](30,4)
		Scope(30,11)
	Function[isProto=0]	Void(33,0)
		Identifier[name:poo](33,5)
		Scope(33,12)
	Function[isProto=0]	Bool(36,0)
		Identifier[name:poo](36,5)
		Scope(36,12)
sema: 5:0 : redefinition of main
sema: 8:0 : redefinition of main
sema: 11:0 : redefinition of main
sema: 15:0 : definition inconsistent with earlier definition of helper1
sema: 24:0 : definition inconsistent with earlier definition of f
sema: 30:0 : redefinition of poo
sema: 33:0 : redefinition of poo
sema: 36:0 : redefinition of poo
